File: src/model/Dot.go
package model

import "github.com/HuXin0817/dots-and-boxes/src/config"

type Dot int

const (
	DotSize     = config.BoardSize + 1
	MaxDot  Dot = DotSize * DotSize
)

func newDot(x, y int) Dot { return Dot(x*DotSize + y) }

func (d Dot) X() int { return int(d) / DotSize }

func (d Dot) Y() int { return int(d) % DotSize }

File: src/model/Dot.cpp
#include "Dot.h"

Dot::Dot(int v) : v(v) {
}

Dot::Dot(int x, int y) : v(x * Size + y) {
}

[[nodiscard]] int
Dot::X() const {
  return v / Size;
}

[[nodiscard]] int
Dot::Y() const {
  return v % Size;
}

Dot::
operator int() const {
  return v;
}